internal/rpcapi/plugin.go

```diff
@@ -91,3 +96,19 @@ func serverHandshake(s *grpc.Server, opts *serviceOpts) func(context.Context, *t
 type serviceOpts struct {
 	experimentsAllowed bool
 }
+
+func newServiceDisco(cliConfig *terraform1.CLIConfig) (*disco.Disco, error) {
+	services := disco.New()
+	credSrc := newCredentialsSource()
+
+	if cliConfig != nil {
+		for host, cred := range cliConfig.GetCredentials() {
+			if err := credSrc.StoreForHost(svchost.Hostname(host), auth.HostCredentialsToken(cred.Token)); err != nil {
+				return nil, fmt.Errorf("problem storing credential for host %s with: %w", host, err)
```

Comment:
Based on the current implementation for `auth.CredentialsSource`, should never yield an error. However, do not want to ignore if we decide to return an error in the future.